File: src/main/java/com/blink/backend/controller/patient/dto/PatientDto.java
package com.blink.backend.controller.patient.dto;

import com.blink.backend.persistence.entity.appointment.Patient;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class PatientDto {
    private String name;
    private String phoneNumber;
    private String clinicName;

    public static PatientDto fromEntity(Patient patient) {
        return PatientDto.builder()
                .name(patient.getName())
                .phoneNumber(patient.getPhoneNumber())
                .clinicName(patient.getClinic().getClinicName())
                .build();
    }
}

File: src/main/java/com/blink/backend/controller/patient/dto/PatientDto.kt
package com.blink.backend.controller.patient.dto

import com.blink.backend.persistence.entity.appointment.PatientEntity
import java.util.UUID

data class PatientDto(
    val code: UUID,
    val name: String,
    val phoneNumber: String,
    val clinicCode: String
) {
    companion object {
        fun fromEntity(patient: PatientEntity) = PatientDto(
            code = patient.code,
            name = patient.name,
            phoneNumber = patient.phoneNumber,
            clinicCode = patient.clinic.code
        )
    }
}
